### PR TITLE
Add base presenter and alternative disposable observer

### PR DESCRIPTION
**Pull request by albodelu**

### CODE REVIEW

File: app/src/main/kotlin/com/fernandocejas/sample/features/base/BasePresenter.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.fernandocejas.sample.features.base
+
+abstract class BasePresenter<V : BaseView> : Presenter<V>() {
+
+    override fun disposeView() = view.dispose()
+
+    override fun dropView() = super.dropView()
```
[the remark this answers is not in the document]
Comment (albodelu) [reply app/src/main/kotlin/com/fernandocejas/sample/features/base/BasePresenter.kt]: My app is not a sample and I have unrelated code here, I only deleted that line.

File: app/src/main/kotlin/com/fernandocejas/sample/features/base/BasePresenter.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    override fun takeView(view: V) = super.takeView(view)
```
[the remark this answers is not in the document]
Comment (albodelu) [reply app/src/main/kotlin/com/fernandocejas/sample/features/base/BasePresenter.kt]: Same here

File: app/src/main/kotlin/com/fernandocejas/sample/features/base/BasePresenter.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (anstaendig) [reply app/src/main/kotlin/com/fernandocejas/sample/features/base/BasePresenter.kt]: got it

**Comment by android10** [comment]
This project is still a WIP and it is very likely to change. It is coming soon and afterwards we can send PRs. For now I'm going to close this one and check the changes as feedback to continue making progress. 

Thanks!

**Pull request closed without merging**